### PR TITLE
Associate area and data path with lens.

### PR DESCRIPTION
**Pull request by kisoso**
We need the area and data path to be associated with a lens.

Linking them to the label is a cheap way to obtain it as each sit on a lens detail which is on a lens. The link to `vm:forThing` also remains. We kind of follow the building structure of the semantic panel where those semantics exist alongside labels position/display.
We also need to pull geo data from the last detail level (once per lens).

Another way which is more descriptive would be each of the area and data path to be there own class.
Example:
```python
    transform.setdefault('conds', {}).update({
        'area_iri': ('{row[area].exists}', 'vm:s{row[layer].as_text}-{row[ident].as_text}', None),
        'data_iri': ('{row[dataLocation].exists}', 'vm:d{row[layer].as_text}-{row[ident].as_text}', None),
    })
    transform.setdefault('triples', []).extend([
        ('{area_iri}', 'rdf:type', 'vm:Poly'),
        ('{area_iri}', 'vm:forThing', for_prop),
        ('{area_iri}', 'vm:forLens', '{row[lens].as_text}'),
        ('{area_iri}', 'vm:atGeoPoly', '{row[area].as_text}'),
        ('{data_iri}', 'rdf:type', 'vm:DataPath'),
        ('{data_iri}', 'vm:forThing', for_prop),
        ('{data_iri}', 'vm:forLens', '{row[lens].as_text}'),
        ('{data_iri}', 'vm:withGeoPath', '{row[dataLocation].as_text}'),
    ])
    transform['allow-empty-subject'] = True
```

[JIRA](https://visual-meaning.atlassian.net/browse/SMP-384?atlOrigin=eyJpIjoiMzA5Njc2YjIwOTJhNGM3OWE0NWIzYWU5MDlkNzVmZGQiLCJwIjoiaiJ9)

### CODE REVIEW

**Review by alightwing — COMMENTED** [review]
If I have a map with 3 detail levels, and 3 labels for the same thing on each of those detail levels, then the _lowest_ (as in, most-zoomed-in) detail level label gets the `area` and `dataLocation` properties. And then the client does something clever to retrieve this info to get a lens-specific location for that thing. What does the pan + zoom behaviour for focusing then look like? Since this info comes from the last detail level on which an item appears, does it zoom all the way in to this detail level?

**Comment by kisoso** [comment]
Currently the illustrator outputs area for all of the detail levels (if provided(?)).
The zoom-pan focusing prioritizes the area, which is irrespective of detail level, over other location. Area bound fits the screen which might change detail level (e.g. when zoom is linked to detail level).
There is a limitation put on the map creator that the area of only one of the detail levels (the lowest) would be used, although not validated. It is a question if we want to have that limitation reflected in this step.
In any case, both area and data path would be most ideally linked to detail level as it contains lens info as well. The client may have more options to deal with it.

**Review by mariya-gfx — APPROVED** [review]
Thanks, looks sensible.

File: labels.py [hunk 1]
```diff
@@ -239,11 +239,13 @@ def _recursive_labels(semantics, scaler, dlevel_props, parent, defaults):
         )
         o['text'] = deriveText(o['contents'])
         o['anchorName'] = semantic.get('anchorName', o['note'].pop('anchorName', parent['name'].get('anchorName', '')))
+        area = deriveLocation(scaler, semantic.get('area', semantic['visibleBounds'])) if is_last_dlevel else None
```
[the remark this answers is not in the document]
Comment (kisoso) [reply labels.py:242]: None will result in skipping the key.

[the remark this answers is not in the document]
Comment (mariya-gfx) [reply labels.py:242]: so `vm:atGeoPoly` and `vm:withGeoPath` will be `{}`?

**Review by alightwing — COMMENTED** [review]
Last question: `area` is indeed invariant across detail level, but is the same true of `dataCollections`? Surely pink lines are going to move location depending on how zoomed-in a user is and how the positions and sizes of labels change, and thus we need a `dataCollections` property that's unique to each label? See for example EWR, where the pink lines shift per detail level to follow the label resizing.

**Review by alightwing — APPROVED** [review]
I've checked with Mark, and the outcome is that data collection lines that are specific to an instance of a label are an antipattern that should be avoided. Also the EWR example I gave is actually automatically generated data collection lines. I think you're good to go ahead and merge. :)

**Pull request merged**